src/OfflineImageStore.js

```diff
@@ -297,10 +297,14 @@ class OfflineImageStore {
         if(status >= 400)
           return this._removeIfExists(tempPath)
             .then(() => {
-              throw res.text();
+              throw new Error(`Image was downloaded with status code ${status}`);
             });
-        if(existingPath)
-          return this._removeIfExists(existingPath);
+        const promises = [];
+        promises.push(this._removeIfExists(path));
+        if(existingPath) {
+          promises.push(this._removeIfExists(existingPath));
+        }
+        return Promise.all(promises);
```

Comment:
In addition to `existingPath` also remove `path` - there are edge cases when data about cached images and images themselves are out of sync, and since `RNFetchBlob.fs.mv(src, dest)` fails if destination is an existing file, this prevents such errors. (Happened to me in development when I force-cleared iOS storage, but left cached images unchanged).